airbyte-ci/connectors/pipelines/pipelines/actions/environments.py

```diff
@@ -470,14 +471,30 @@ def with_global_dockerd_service(dagger_client: Client) -> Container:
         Container: The container running dockerd as a service
     """
     return (
-        dagger_client.container()
-        .from_(consts.DOCKER_DIND_IMAGE)
-        .with_mounted_cache(
-            "/tmp",
-            dagger_client.cache_volume("shared-tmp"),
-        )
+        dagger_client.container().from_(consts.DOCKER_DIND_IMAGE)
+        # We set this env var because we need to use a non-default zombie reaper setting.
+        # The reason for this is that by default it will want to set its parent process ID to 1 when reaping.
+        # This won't be possible because of container-ception: dind is running inside the dagger engine.
+        # See https://github.com/krallin/tini#subreaping for details.
+        .with_env_variable("TINI_SUBREAPER", "")
+        # Expose the docker host port.
         .with_exposed_port(2375)
-        .with_exec(["dockerd", "--log-level=error", "--host=tcp://0.0.0.0:2375", "--tls=false"], insecure_root_capabilities=True)
+        # Mount the docker cache volumes.
+        .with_mounted_cache("/tmp", dagger_client.cache_volume("shared-tmp"), sharing=CacheSharingMode.PRIVATE)
+        # Mount /var/lib/docker so docker writes to the host file system, i.e. the dagger engine.
+        .with_mounted_cache("/var/lib/docker", dagger_client.cache_volume("shared-var-lib-docker"), sharing=CacheSharingMode.PRIVATE)
```

Comment:
It's still not clear to me what `PRIVATE` cache sharing mode means.
In the doc it is stated that the volume will only be used for the time of a pipeline. But I don't know what they mean by pipeline: a dagger session or a `dagger_client.pipeline().container()`... Definitely something to deblur with the Dagger team.

I'm not against using it otherwise, I think it helps the "temporary" caching use case you need, but I'm a bit concerned about the side effects as I'm not understanding the exact behavior.

Reply:
You're right and it turns out this is bad and wrong. The docker service is "global" in that it's created outside of all pipelines. Will revert.